# Exploratory Data Analysis

## Objective
- Perform first real business analysis

## Import Library and set up data

In [18]:
import pandas as pd

df = pd.read_csv("../data/processed/retail_sales_cleaned.csv")

df.head()

,Transaction ID,Date,Customer ID,Gender,Age,Product Category,Quantity,Price per Unit,Total Amount
0,1,2023-11-24,CUST001,Male,34,Beauty,3,50,150
1,2,2023-02-27,CUST002,Female,26,Clothing,2,500,1000
2,3,2023-01-13,CUST003,Male,50,Electronics,1,30,30
3,4,2023-05-21,CUST004,Male,37,Clothing,1,500,500
4,5,2023-05-06,CUST005,Male,30,Beauty,2,50,100


## What is the Total Revenue?

In [19]:
df["Total Amount"].sum()

456000

### Business Insight
- Dataset reveals that retail has total revenue of 456,000

## Which Product Category generates the highest revenue?

In [20]:
df.groupby("Product Category")["Total Amount"].sum()

Product Category
Beauty         143515
Clothing       155580
Electronics    156905
Name: Total Amount, dtype: int64

### Business Insight
- Electronics Category has highest reveue
- Electronis has reveune of 156905
- Clothing comes in seconde for the revenue


## Which Product Category has most transactions?

In [21]:
df["Product Category"].value_counts()

Product Category
Clothing       351
Electronics    342
Beauty         307
Name: count, dtype: int64

### Business Insight
- Clothing has most transaction in dataset
- With total 351 transactions on it

## Average Customer Age

In [22]:
df["Age"].mean()

41.392

### Business Insight
- The average age of customer is approximately 41.392 
- Which could mean older individual is average customer for our retail


## Gender Distribution of Customers

In [23]:
df["Gender"].value_counts()

Gender
Female    510
Male      490
Name: count, dtype: int64

### Business insights
- Our retail dataset has higher female customer distribution
- 20 more female customer is there compared to male 
- 